### PR TITLE
[MUS-61] Basic header

### PR DESCRIPTION
**Pull request by nickkuchko**
not Logged in
Logged in

### CODE REVIEW

**Review by Drelarusoidd — APPROVED** [review]

File: src/components/navBar.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import { Link } from 'react-router-dom';
+import AudiotrackRoundedIcon from '@mui/icons-material/AudiotrackRounded';
+import { useState, useEffect } from 'react';
+
+
+export default function NavBar() {
+    const [isLogin, SetIsLogin] = useState(false);
+
+    useEffect(() => {
+        let token = localStorage.getItem('access');
```
[the remark this answers is not in the document]
Comment (nickkuchko) [reply src/components/navBar.js]: Done

File: src/components/navBar.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import Typography from '@mui/material/Typography';
+import Button from '@mui/material/Button';
+import IconButton from '@mui/material/IconButton';
+import { Link } from 'react-router-dom';
+import AudiotrackRoundedIcon from '@mui/icons-material/AudiotrackRounded';
+import { useState, useEffect } from 'react';
+
+
+export default function NavBar() {
+    const [isLogin, SetIsLogin] = useState(false);
```
Comment (ZubelRuslan) [inline src/components/navBar.js:14]: better to use traditional camelCase

File: src/components/navBar.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                            <AudiotrackRoundedIcon />   {/* just a clickable icon. Assign menu items later */}  
+                        </IconButton>
+                        <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
+                            <Button component={Link} to="/" color="inherit">Musichub</Button>
+                        </Typography>
+                        <Button component={Link} to="/login" color="inherit">Login</Button>
+                        <Button component={Link} to="/register" color="inherit">Create account</Button>
+                    </Toolbar>
+                </AppBar>
+            </Box>
```
Comment (ZubelRuslan) [inline src/components/navBar.js:70]: A lot of code duplication

File: src/containers/AuthContainer.js [hunk 4]
```diff
@@ -4,13 +4,17 @@ import { signIn } from "../services/AuthService";
...
 
 const AuthContainer = () => {
     const [username, setUsername] = useState("")
     const [password, setPassword] = useState("")
+    const navigate = useNavigate()
 
     const Auth = async () => {
        await signIn(username, password)
+       navigate('/');
+       window.location.reload(); // need to update navbar menu items after log in
```
Comment (ZubelRuslan) [inline src/containers/AuthContainer.js:17]: This global state should be stored in the scope of some root element. Root element will store this data and send it to child component. In general it's a good case to use React Context or Redux

File: src/containers/LogoutContainer.js [hunk 5]
```diff
@@ -13,6 +13,7 @@ const Logout = () => {
     useEffect (() => {
         if (LogoutButton()) {
             navigate('/');
+            window.location.reload(); // need to update navbar menu items after logout
```
Comment (ZubelRuslan) [inline src/containers/LogoutContainer.js:16]: Same

**Review by ZubelRuslan — CHANGES_REQUESTED** [review]
Before merge nav bar jsx should be refactored

File: src/components/navBar.js (hunk 3 above)
Comment (ZubelRuslan) [reply src/components/navBar.js:70]: Main mind of comments for this MR is remove this code duplication

Comment (nickkuchko) [reply src/components/navBar.js:70]: I did that already, now I'm trying to figure out Context, I think I did everything, but the buttons don't update :)

Comment (ZubelRuslan) [reply src/components/navBar.js]: @nickkuchko you can push latest changes and mention me in PR comment to check

Comment (nickkuchko) [reply src/components/navBar.js]: @[ZubelRuslan](https://github.com/ZubelRuslan) Hi, I've just made an update using React Context

File: src/components/navBar.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+                <AppBar position="static">
+                    <Toolbar>
+                        <MainNavbuttons />
+                        <Button component={Link} to="/login" color="inherit">Login</Button>
+                        <Button component={Link} to="/register" color="inherit">Create account</Button>
+                    </Toolbar>
+                </AppBar>
+            </Box>
+        )
+    }
```
Comment (ZubelRuslan) [inline src/components/navBar.js:41]:
```suggestion
        return (
            <Box sx={{ flexGrow: 1 }}>
                <AppBar position="static">
                    <Toolbar>
                        <MainNavbuttons />
                        {{ isLogin ? 
                        <Button component={Link} to="/profile" color="inherit"> {user}'s Profile </Button>
                        <Button component={Link} to="/logout" color="inherit">Logout</Button>
                        :
                        <Button component={Link} to="/login" color="inherit">Login</Button>
                        <Button component={Link} to="/register" color="inherit">Create account</Button>
                        }}
                </Toolbar>
            </AppBar>
            </Box >
        );
    }
```

Comment (ZubelRuslan) [reply src/components/navBar.js:41]: I'm already told you about duplication...

File: src/containers/AuthContainer.js [hunk 7]
```diff
@@ -1,51 +1,68 @@
...
     const Auth = async () => {
-       await signIn(username, password)
+        await signIn(username, password)
+        setUser(username)
+        const token = localStorage.getItem('access');
+        if (token) {
+            setIsLogin(true);
+        } else {
+            setIsLogin(false);
+        }
```
Comment (ZubelRuslan) [inline src/containers/AuthContainer.js:25]:
```suggestion
   setIsLogin(!!token);
```

**Review by ZubelRuslan — CHANGES_REQUESTED** [review]

File: src/containers/AuthContainer.js (hunk 7 above)
Comment (nickkuchko) [reply src/containers/AuthContainer.js]: @[ZubelRuslan](https://github.com/ZubelRuslan) Hi, I've updated requested files

File: src/components/navBar.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    const { user } = useContext(Context)
+    const { isLogin } = useContext(Context)
+
+    return (
+        <Box sx={{ flexGrow: 1 }}>
+            <AppBar position="static">
+                <Toolbar>
+                    <MainNavbuttons />
+                    { isLogin ? 
+                    <div>
```
Comment (ZubelRuslan) [inline src/components/navBar.js:22]: btw you can use fragment here to avoid unnecessary `div`

**Review by ZubelRuslan — APPROVED** [review]

**Pull request merged**